### PR TITLE
Network policy arch for engines

### PR DESCRIPTION
**Pull request by naumvd95**
- network-policy methods
- resizing method refactor due dependencies

### CODE REVIEW

File: kqueen_ui/blueprints/ui/views.py [hunk 1]
```diff
@@ -700,17 +700,74 @@ def handle(self, cluster_id):
...
+                                            url_for('ui.index')))
+
+
+class ClusterSetNetworkPolicy(KQueenView):
+    decorators = [login_required]
+    methods = ['POST']
+    validation_hint = 'uuid'
+
+    def handle(self, cluster_id):
+        unsupported_instances = ['g1-small', 'f1-micro']
```
[the remark this answers is not in the document]
Comment (naumvd95) [reply kqueen_ui/blueprints/ui/views.py]: fixed

File: kqueen_ui/blueprints/ui/views.py [hunk 2]
```diff
@@ -700,17 +700,74 @@ def handle(self, cluster_id):
...
+        cluster = self.kqueen_request('cluster', 'get', fnargs=(cluster_id,))
+
+        network_provider = request.form['network_provider']
```
[the remark this answers is not in the document]
Comment (naumvd95) [reply kqueen_ui/blueprints/ui/views.py]: @ekhomyakova can u help me with that?

**Comment by naumvd95** [comment]
using as base for @ekhomyakova

File: kqueen_ui/blueprints/ui/views.py [hunk 3]
```diff
@@ -700,17 +700,83 @@ def handle(self, cluster_id):
...
-            flash("{} engine doesn't support scaling.".format(prettify_engine_name(engine)), 'warning')
-            return redirect(request.environ.get('HTTP_REFERER', url_for('ui.index')))
+            flash("{} engine doesn't support scaling."
+                  .format(prettify_engine_name(engine)), 'warning')
+            return redirect(request.environ.get('HTTP_REFERER',
+                                                url_for('ui.index')))
         current_node_count = cluster['metadata']['node_count']
         node_count = request.form['node_count']
+
+        if int(node_count) < 2 and cluster['networkPolicy']['enabled'] is True:
```
Comment (atengler) [inline kqueen_ui/blueprints/ui/views.py]: I would still prefer to create a way to check available features through API call to backend, but if you want to check in UI, you still need to take into account that not all clusters have `networkPolicy` key, so it should be - `cluster.get('networkPolicy', {}).get('enabled', False)` instead. Also I thought its under `metadata` key, not a top level attribute of cluster? If so then its - `cluster.get('metadata', {}).get('networkPolicy', {}).get('enabled', False)`

**Pull request closed without merging**